src/qtt/measurements/scans.py

```diff
@@ -851,7 +850,7 @@ def _calculate_sweepvalues(param, sweep_data, inclusive_end=False) -> SweepFixed
                 raise ValueError('step value may not be 0')
 
             if inclusive_end:
-                epsilon = 1e-8
+                epsilon = np.sign(sweep_data['step'])*1e-8
```

Comment:
Nice catch. It would be great when you add the minimal example given in the description of the pr as an succeeding test in unittests/measurements/test_scans.py